src/psyclone/tests/generator_test.py

```diff
@@ -780,6 +781,16 @@ def test_main_api():
     main([filename, "-api", "dynamo0.3"])
     assert Config.get().api == "lfric"
 
+    # Test we get the logging debug correctly with caplog. This
+    # overrides the file output that PSyclone attempts.
+    caplog.clear()
+    with caplog.at_level(logging.DEBUG):
+        main([filename, "-api", "dynamo0.3", "--logging", "debug",
+              "--logfile", "test.out"])
```

Comment:
Maybe add a test that without the flags we get the expected values and a test with an invalid value to see that we get the right error message.

Also, is there a way to retrieve the logfile from caplog? This would be enough to check it has the command line flag. We don't really need to create files if is problematic with pytest, as we can trust python builtins.

Reply:
Added tests for this. 
Only thing I can't do is check the output file, I've not managed to find any way to do that inside pytest.

Reply:
I don't know that much about the logger, but for the driver creation: I change into tmpdir, and then read the file (created  in $PWD). Not sure if that helps, just thought to mention it ;)